### DISCUSSION TITLE
Aggretation Time not working as intended

### DISCUSSION THREAD

**Discussion by P1514** [jertel/elastalert2#800] [Q&A] [answered]
Hi,
Configuration:
run_every: 30s
aggregation: 2m

Situation is if the query takes 10h to get all data (mass flood of matches, or old data rerun) the aggregation time seems to be ignored and only when the run finished it will attempt to send it.

I think this might be related with the way it gets aggregated id to run from elastsearch (find_pending_aggregate_alert)

Any one else had this issue?

**Comment by jertel** [comment, marked as answer]
ElastAlert 2 expects queries to complete very quickly, e.g. elapsed time measured in seconds or even sub-seconds. A query that takes 10 hours to complete is not going to work well with this application.

**Comment by P1514** [comment]
Sure I agree with you 10h long queries are inefficient. My question is more about, lets say i want to run for the past 30 days.
rule takes 2 hours to run for the past 30 days on a 10m buffer time.
If I don't use aggregate it will send alert each time a match is found.
If i use aggregate (5 minutes), it will keep this and log (next run at 16:00) for the entire 2h hours even though the next run time has already passed. 
So even though my rule normally should not take more than sub-seconds, if some anomaly occurs that will make them run for 10h. Rule will not alert until all the 10h have passed.

I was looking through to try and understand why this happens but I couldn't see anything besides the run_rule keeping using the same aggregate id even though it had already passed the time

**Reply by jertel** [reply]
It's not entirely clear what you're asking but it sounds as if you're looking for ElastAlert 2 to send partial alert results while a long query is running? If the app is busy waiting for Elastic to respond then its program flow for that rule is completely blocked until the response returns. 

The following discussions might help explain how bad query performance impacts timing:

- https://github.com/jertel/elastalert2/issues/187
- https://github.com/jertel/elastalert2/discussions/456
